### ISSUE TITLE
Add environment variable to disable spinner output

### ISSUE THREAD

**Issue by leonardobiffi** [infracost/infracost#1450]
Suggestion to not have spinner output when setting a variable like `INFRACOST_NO_SPINNER=true`

Example to avoid in Gitlab pipeline:
![image](https://user-images.githubusercontent.com/20444705/157677764-e3d1f45e-10dc-4c55-9511-665b438ff5bc.png)

**Comment by aliscott** [comment]
Thanks @leonardobiffi. I'm a plus one on this definitely. Terraform has a `TF_IN_AUTOMATION` variable, maybe we can do something similar. In the meantime you can set the `INFRACOST_LOG_LEVEL=info` env variable to avoid this.

**Comment by leonardobiffi** [comment]
Hello @aliscott, I submitted this PR with a suggestion to disable the spinner. Thanks

**Comment by aliscott** [comment]
Thanks @leonardobiffi! I want to take a little more time to think about this and get feedback from others. There's other places in the code where we have different functionality for CI runs, i.e. messaging, caching. We currently do this using [this method](https://github.com/infracost/infracost/blob/b987f6d6dc036629c2655da708faeac0ff193150/internal/config/run_context.go#L124).

What are your thoughts on if we want an env variable for all these cases, or if we should detect automatically?

**Comment by leonardobiffi** [comment]
@aliscott I believe that the spinner output can really be something disabled by default when executing by CI. For my use case it would be great.

**Comment by aliscott** [comment]
@leonardobiffi just to update you that we've brought it on our sprint to look at this solution alongside other options. Will keep you updated.

**Comment by vdmgolub** [comment]
Here are the solutions that we can implement to deal with spinners issue in CI, from easiest:

1. The spinners are automatically disabled when logging is enabled, they are replaced with text messages like `Starting: X`/`Completed: X`. Setting `INFRACOST_LOG_LEVEL=info` env var is the simplest way to do that.
   - Pros:
     - No need to change CLI code.
   - Cons:
     - Manual step to use it required from users.
     - Need to update all Infracost CI examples to use the env var.
     - Not straightforward and cryptic usage.
2. Add a dedicated `INFRACOST_NO_SPINNER` to completely disable the spinners and messaging.
   - Pros:
     - A dedicated env var for the feature. Clear usage.
     - The feature is already implemented and ready for review: #1451.
   - Cons:
     - Manual step to use it required from users
     - Need to modify the CLI code to support the env var.
     - Need to update all Infracost CI examples to use the env var.
     - Need to update docs to mention the env var usage.
     - Too granular, using this approach we can end up with too many env vars to tweak the CLI behavior.
3. Add a new `INFRACOST_CI` or `INFRACOST_IN_AUTOMATION` env var, use it to completely disable the spinners.
   - Pros:
     - A dedicated env var for CI. May be used for other settings in future.
   - Cons:
     - Manual step to use it required from users
     - Need to modify the CLI code to support the env var.
     - Need to update all Infracost CI examples to use the env var.
     - Need to update docs to mention the env var usage.
4. Autodetect CI environments (we already have `IsCIRun` in the code) and disable the spinner if that’s true:
   - Pros:
     - Default behavior, no need to edit CI by users.
     - Easy code change.
   - Cons:
     - No way to enable it in CI. But does anyone need it in CI anyway?
5. Remove spinners:
   - Pros:
     - Default behavior.
   - Cons:
     - No visibility what CLI does if running locally.
     - No visibility when logging is enabled.

Open question: should we completely remove spinners from CI, or should we still keep text messages (like when logging is enabled)?

@leonardobiffi @aliscott @alikhajeh1 @hugorut @tim775 This is what we came up with. What do you think?

I personally prefer option #4 as it is automatic, doesn't need much explanation and streamlines CI case.

**Comment by leonardobiffi** [comment]
I liked option 4 too.  I believe the spinner output in CI has no need

**Comment by hugorut** [comment]
I'd be pro option 5. We can just replace the spinners with just generic log lines. This would solve the duplication issue in CI and still give the users information about what is happening.

I'm also happy with option 4 though.

**Comment by aliscott** [comment]
Thanks @vdmgolub for explaining the different options. Here's my initial thoughts.

Option 3: This is good because we can use it in our CI integrations and it means users can add this for any CI pipeline, even if we don't support it. This is also consistent with how Terraform does it.

Option 4: Simple, but doesn't have a fallback for users who's CI platform we don't recognise

Option 5: I like this, but not sure if it fully solves our problem. There's other things we may want to turn off in CI, such as colors.

In summary: I think option 4 is good, but can see the need for also having 3 to provide a fallback.

**Comment by vdmgolub** [comment]
Thank you everyone! After a brief talk with Alistair we realized the easiest solution is setting log level to `info` when Infracost is running in CI. We already do that for our [GH Actions](https://github.com/infracost/actions/blob/master/setup/src/index.ts#L119), it makes sense to have such behavior for all platforms.

Here's the PR: https://github.com/infracost/infracost/pull/1474.

We will most likely revisit how we deal with logging and output in future, but until we know more it should be enough for now.

@leonardobiffi Thank you very much for creating this issue and pointing us to the problem! Community's feedback is essential to keep improving and adding more cool features to Infracost! ❤️ And thank you for your PR, it looks great, but it seems like we'll have to close it for now.

**Comment by vdmgolub** [comment]
The solution will go live with the next release. We will notify when it's out. Thank you for creating the issue!

**Comment by vdmgolub** [comment]
@leonardobiffi This is live with v0.9.21! Thank you! :)
